### PR TITLE
IRGen: cast the argument type for the OpaqueExtentialTypeInfo assignm…

### PR DESCRIPTION
**Pull request by compnerd**
…ent op

Ensure that the function parameter types match, as this triggers an assertion
failure otherwise.  Thanks to Dave Zarzycki for finding this!

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @davezarzycki

**Comment by compnerd** [comment]
(Note that this is untested)

**Comment by davezarzycki** [comment]
Verified. This fixes the regression and now the whole validation test suite passes. Thanks @compnerd!

**Review by davezarzycki — APPROVED** [review]

File: lib/IRGen/GenExistential.cpp
Comment (CodaFi) [inline lib/IRGen/GenExistential.cpp]: Should these bitcasts be explicitly sequenced outside the construction of this object?

Comment (compnerd) [reply lib/IRGen/GenExistential.cpp]: Damn it!  You would think that I would learn this based on the fact that I'm fixing this exact issue.  You're right @CodaFi.

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Comment by davezarzycki** [comment]
GitHub doesn't seem to be blocking this informally tested branch from being merged. Does Swift CI even work/apply to branches that aren't named "master"?

**Comment by compnerd** [comment]
@davezarzycki - yeah, I never had this work very well for me, which is why I normally wait for someone else to have a look (and merge) the patches ... cause I don't want to accidentally break the branch and I figure a second pair of eyes is helpful.

**Comment by CodaFi** [comment]
Tagging @bob-wilson who works a lot in this area.

**Comment by compnerd** [comment]
Alright, verified that the extraction also was correct.  Going to merge this.

**Comment by bob-wilson** [comment]
I was trying to review this and track down exactly what changed here, but I assume it is something from the opaque pointer types work. It looks sane regardless, so I'll just take your word for it. Thanks!

**Comment by compnerd** [comment]
@bob-wilson - oh, if you're curious, I can explain the root cause here.  This is related to the opaque pointer types work.  The `getOrInsertFunction` method changed to return a `FunctionCallee`.  Additionally, it no longer would return a `Value` that is the `BitCastInst` for the adjusted type signature.  This exposes the original function that we are re-using with the original signature.  The result is that the bitcasting has shifted now from the LLVM IR[Builder] to the Swift side, and pushed down to the generation of the `CallSite`.

**Pull request merged**
